engine/src/main/java/org/entando/entando/aps/system/services/api/cache/ApiCatalogManagerCacheWrapper.java

```diff
@@ -0,0 +1,158 @@
+		} catch (Throwable t) {
+			logger.error("Error bootstrapping ApiCatalog cache", t);
+			throw new ApsSystemException("Error bootstrapping ApiCatalog cache", t);
+		}
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public Map<String, ApiResource> getMasterResources() {
+		return this.get(this.getCache(), APICATALOG_RESOURCES_CACHE_NAME, Map.class);
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public Map<String, ApiService> getMasterServices() {
+		return this.get(this.getCache(), APICATALOG_SERVICES_CACHE_NAME, Map.class);
+	}
+
+	@Override
+	public void addService(ApiService apiService) {
+		this.manage(apiService.getKey(), apiService, Action.ADD);
+	}
+
+	@Override
+	public void updateService(ApiService apiService) {
+		this.manage(apiService.getKey(), apiService, Action.UPDATE);
+	}
+
+	@Override
+	public void removeService(String key) {
+		this.manage(key, null, Action.DELETE);
+	}
+
+	protected void insertObjectsOnCache(Cache cache, Map<String, ApiResource> resources, IApiCatalogDAO apiCatalogDAO) throws ApsSystemException {
+		Map<String, ApiService> services = new HashMap<>();
+		try {
+			List<ApiMethod> apiGETMethods = buildApiGetMethods(resources);
+			services = apiCatalogDAO.loadServices(apiGETMethods);
+
+			for (Map.Entry<String, ApiService> entry : services.entrySet()) {
+				cache.put(APICATALOG_SERVICE_CACHE_NAME_PREFIX + entry.getKey(), entry.getValue());
+			}
+
+		} catch (Throwable t) {
+			services = new HashMap<String, ApiService>();
+			logger.error("Error loading Services definitions", t);
+			throw new ApsSystemException("Error loading Services definitions", t);
+		}
+		cache.put(APICATALOG_RESOURCES_CACHE_NAME, resources);
+		cache.put(APICATALOG_SERVICES_CACHE_NAME, services);
+	}
+
+
+	protected void releaseCachedObjects(Cache cache) {
+		this.releaseCachedObjects(cache, APICATALOG_SERVICES_CACHE_NAME, APICATALOG_SERVICE_CACHE_NAME_PREFIX);
+		cache.evict(APICATALOG_RESOURCES_CACHE_NAME);
+	}
+
+	@SuppressWarnings("unchecked")
+	protected void releaseCachedObjects(Cache cache, String cacheKey, String codePrefix) {
+		Map<String, ApiService> services = (Map<String, ApiService>) this.get(cache, cacheKey, Map.class);
+		if (null != services) {
+			for (Map.Entry<String, ApiService> entry : services.entrySet()) {
+				String key = entry.getKey();
+				cache.evict(codePrefix + key);
+			}
+			cache.evict(cacheKey);
+		}
+	}
+
+	protected List<ApiMethod> buildApiGetMethods(Map<String, ApiResource> resources) {
+		List<ApiMethod> apiGETMethods = new ArrayList<ApiMethod>();
+		List<ApiResource> resourceList = new ArrayList<ApiResource>(resources.values());
+		for (int i = 0; i < resourceList.size(); i++) {
+			ApiResource apiResource = resourceList.get(i);
+			if (null != apiResource.getGetMethod()) {
+				apiGETMethods.add(apiResource.getGetMethod());
+			}
+		}
+		return apiGETMethods;
+	}
+
+	protected void manage(String key, ApiService service, Action operation) {
```

Comment:
@eugeniosant I'm wondering if we are going to want some debug logging here in all of these managers. Not suggesting that we go back and change them now but we have no logging now and we might get punished if we need to troubleshoot something on these in the future. Might be worth adding some debug at least for the big calls (evict, bulk add, init etc)